File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/Ingredient.java
package nl.craftsmen.workshops.reactivemeetup.domain;

public class Ingredient {

    private String name;
    private EIngredientType type;

    public Ingredient(String name, EIngredientType type) {
        this.name = name;
        this.type = type;
    }

    public String getName() { return name; }

    public EIngredientType getType() {
        return type;
    }
}

File: javascript/src/domain/ingredient.js
var method = Ingredient.prototype;

function Ingredient(name, type) {
    this._name = name;
    this._type = type;
}

method.getName = function() {
    return this._name;
};

method.getType = function() {
    return this._type;
}

module.exports = Ingredient;